### PR TITLE
feat(analytics|react): setup promocode applied event next

### PR DESCRIPTION
**Pull request by talbertosilva**
## Description

BREAKING CHANGE:
- The omnitracking integration will no longer remap the
"orderId" parameter on checkout events in case its value is not a valid
order code and will log an error instead of a warning, in
case it is invalid.

Other changes:
- Added `checkoutOrderId` and `orderCode` parameters to promo code
applied event for Omnitracking on next.
- Added `transaction_id` parameter to `apply_promo_code` event for
GA4 on next.

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [ ] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/analytics/src/integrations/Omnitracking/__tests__/omnitracking-helper.test.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('getPlatformSpecificParameters', () => {
 });
 
 describe('getCheckoutEventGenericProperties', () => {
-  it('should display some warn', () => {
+  it.skip('should display some warn', () => {
```
Comment (boliveira) [inline packages/analytics/src/integrations/Omnitracking/__tests__/omnitracking-helper.test.ts:68]: Is this intended?

File: packages/analytics/src/integrations/Omnitracking/__tests__/__snapshots__/Omnitracking.test.ts.snap [hunk 2]
```diff
@@ -174,8 +176,10 @@ Array [
 
 exports[`Omnitracking track events definitions \`Promocode Applied\` return should match the snapshot 1`] = `
 Object {
+  "checkoutOrderId": undefined,
```
Comment (boliveira) [inline packages/analytics/src/integrations/Omnitracking/__tests__/__snapshots__/Omnitracking.test.ts.snap:179]: Shouldn't this be set, according to the PR description?

File: packages/analytics/src/integrations/Omnitracking/__tests__/omnitracking-helper.test.ts (hunk 1 above)
Comment (talbertosilva) [reply packages/analytics/src/integrations/Omnitracking/__tests__/omnitracking-helper.test.ts:68]: The warn tested here was removed from the method so this test got no purpose, I will remove the test

File: packages/react/src/analytics/integrations/GA4/eventMapping.ts [hunk 3]
```diff
@@ -370,6 +370,26 @@ const getCheckoutShippingStepParametersFromEvent = (
...
+ * @returns Properties formatted for the GA4's promocode applied event.
+ */
+const getPromocodeAppliedParametersFromEvent = (
+  eventProperties: EventProperties,
+) => {
+  return {
+    currency: eventProperties.currency,
+    coupon: eventProperties.coupon,
+    value: eventProperties.total,
+    transaction_id: isNaN(Number(eventProperties.orderId))
```
Comment (boliveira) [inline packages/react/src/analytics/integrations/GA4/eventMapping.ts:387]: Can you explain what is the intention here?

**Review by boliveira — APPROVED** [review]

**Review by ruifcnunes — APPROVED** [review]

**Pull request merged**